documentation/projects/proposals/trust_and_safety/content_report_moderation/20240122-implementation_plan_bulk_moderation_actions.md

```diff
@@ -0,0 +1,526 @@
+  particular creator's record but realize upon further investigation that only
+  some of them needed to be deindexed.
+- Undoing _all_ deindexing/marking sensitive for a particular creator, including
+  ones that were originally acted upon by individual reports.
+
+### A DAG for bulk decisions
+
+It's difficult to anticipate all the ways that we might need to filter records
+for selecting a set to take action on. Moreover, filtering in the Django admin
+views (especially by text search terms) may be extremely slow for some use
+cases.
+
+The project proposal's requirements indicate that we need to be able to make
+bulk decisions with selections from admin views. However, I considered adding a
+DAG in addition to the custom actions that would allow for maximum flexibility.
+The DAG which would take an arbitrary SELECT query for selecting records to add
+to a new ModerationDecision. This DAG could even be modified to query
+elasticsearch for performance improvements, and to take advantage of the
+stemming behavior in search.
+
+However, this duplicates some logic and may be excessive optimization for now.
+It would also prevent us from easily extending permission to moderators if
+desired. A DAG may be considered for future optimization if necessary.
+
+## Design
+
+<!-- Note any design requirements for this plan. -->
+
+None, we'll stick with Django admin default styles.
+
+## Blockers
+
+<!-- What hard blockers exist that prevent further work on this project? -->
+
+We must first resolve the overlap with the
+[related IP](https://docs.openverse.org/projects/proposals/trust_and_safety/content_report_moderation/20231208-implementation_plan_django_admin_moderator_access.html).
+
+## Rollback
+
+<!-- How do we roll back this solution in the event of failure? Are there any steps that can not easily be rolled back? -->
+
+Nothing that can't be easily rolled back, except the changes to
+ModerationDecision relative to the original plan. This IP does not require
+making changes to any other models.
+
+## Risks
+
+<!-- What risks are we taking with this solution? Are there risks that once taken can’t be undone?-->
+
+This work allows for bulk actions which would have dramatic effects, like for
+example deindexing an extremely large number of records. When that happens, we
+can reindex but we still need to do a data refresh afterward to get the data
+back, which may take several days.
+
+This is mitigated by tightly restricting access and adding a confirmation page
+that reiterates the number of records that are about to be affected by a
+decision, so a maintainer does not accidentally deindex much more than intended.
+We _could_ also put a limit to the number of records that can be deindexed in a
+single moderation decision as an additional precaution; however, this new bulk
+action does not add any _new_ risk compared to the existing bulk delete actions
+so I do not think this is necessary.
```

Comment:
Ah, you answered my questions! Sounds good to me to not include a new limitation on the max number of records :+1: